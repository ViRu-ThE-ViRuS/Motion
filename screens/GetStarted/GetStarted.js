import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
} from 'react-native';

import { NavigationActions } from 'react-navigation';

import SvgUri from 'react-native-svg-uri';
import PrimaryFlatButton from './../../components/Button/PrimaryFlatButton';

export default class GetStarted extends Component {
    static navigationOptions = {
        header: null,
    }

    startIntro() {
        this.props.navigation.navigate('intro');
    }

    render() {
        return (
          <View style={styles.container}>
            <View style={styles.center}>
              <SvgUri
                        source={require('./../res/motion_logo.svg')}
                        height="150"
                        width="150"
                    />
              <Text style={styles.heading}>Motion</Text>
              <Text style={styles.subHeading}>"create change"</Text>
            </View>
            <PrimaryFlatButton text="Get Started" onPress={() => this.startIntro()} />
          </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'space-between',
        flex: 1,
    },
    heading: {
        fontSize: 35,
        color: '#000',
        fontFamily: 'SourceSansPro_Bold',
    },
    subHeading: {
        fontSize: 20,
        color: '#444',
        fontFamily: 'SourceSansPro_Italic',
    },
    center: {
        marginTop: 150,
        justifyContent: 'center',
        alignItems: 'center',
    },
});
