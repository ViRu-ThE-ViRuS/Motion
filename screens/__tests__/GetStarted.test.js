import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import GetStarted from './../GetStarted/GetStarted';

beforeEach(() => {
    global.fetch = jest.fn((props) => ({
        text: jest.fn(),
    }));
});

it('GetStarted screen renders correctly', () => {
    const mockNavigation = {
        navigate: jest.fn((route) => true),
    };

    const tree = renderer.create(
      <GetStarted navigation={mockNavigation} />
    );

    expect(tree).toMatchSnapshot();
    expect(tree.getInstance().startIntro()).toBe(undefined);
});
