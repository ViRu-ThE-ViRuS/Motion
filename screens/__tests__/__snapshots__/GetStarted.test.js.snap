// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStarted screen tests renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": 150,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "SourceSansPro_Bold",
          "fontSize": 35,
        }
      }
    >
      Motion
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#444",
          "fontFamily": "SourceSansPro_Italic",
          "fontSize": 20,
        }
      }
    >
      "create change"
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#5C7D87",
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontFamily": "SourceSansPro_Regular",
            "fontSize": 15,
          }
        }
      >
        Get Started
      </Text>
    </View>
  </View>
</View>
`;
